kart/apply.py

```diff
@@ -178,6 +186,59 @@ def _build_signature(patch_metadata, person, repo):
     return repo.author_signature(**signature)
 
 
+def parse_file_diff(file_diff_input, allow_minimal_updates=None):
+    def convert_half_delta(half_delta):
+        if half_delta is None:
+            return None
+        elif half_delta.value.startswith("base64:"):
+            return (half_delta.key, b64decode_str(half_delta.value))
+        else:
```

Comment:
```
"+": "base64:\nbase64 is a way of encoding binary data in strings that you can use in JSON, like this patch."
```

to keep the encoding prefix purely unambiguous maybe we need an enforced prefix for text content? maybe just `:` or `text:`. Ugly though. But it'd be annoying to have a list of prefixes you're not allowed in text data.

Reply:
Yeah I did this prefix thing a bit late and it's a bit ugly - again, it's because there's not really any room for anything out-of-band in the JSON diffs. JSON-LINES on the other hand has more room and it has a "binary" attribute that can disambiguate these two cases. I was hoping we'd just switch to something more like JSON-LINES for everything including patches but we're not really ready for that... this will work for now, until and unless someone starts creating patches which modify text files to have "base64:" at the start of them. It will go away if we ever manage to drop the JSON format.

Reply:
Modified the patch format for text files to start with "text:", but left the other diffs as they were.